src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3904,21 +3939,36 @@ export class ResultsViewPageStore {
         },
     });
 
-    readonly treatmentLinkMap = remoteData<{ [treatmentId: string]: string }>({
-        await: () => [this.molecularProfilesInStudies],
+    readonly genericAssayEntitiesGroupByGenericAssayTypeLinkMap = remoteData<{
+        [genericAssayType: string]: { [stableId: string]: string };
+    }>({
+        await: () => [this.genericAssayEntitiesGroupByGenericAssayType],
         invoke: async () => {
+            const linkMapGroupByGenericAssayType: {
+                [genericAssayType: string]: { [stableId: string]: string };
+            } = {};
             if (
-                this.molecularProfilesInStudies.result &&
-                this.molecularProfilesInStudies.result.length > 0
+                !_.isEmpty(
+                    this.genericAssayEntitiesGroupByGenericAssayType.result
+                )
             ) {
-                const treatments = await fetchTreatmentByMolecularProfileIds(
-                    this.molecularProfilesInStudies.result
+                const genericAssayTypes = _.keys(
```

Comment:
i think the following code can be rewritten more simply as mapValues on genericAssayEntitiesGroupByGenericAssayType.result